packages/web/src/uiUtils.ts

```diff
@@ -165,6 +165,12 @@ export function useSetting<S>(name: string | undefined, defaultValue: S, title?:
   return [value, setValueWrapper, setting];
 }
 
+declare global {
+  interface Window {
+    saveSettings(): Promise<void>;
```

Comment:
```suggestion
    saveSettings?(): Promise<void>;
```

A couple lines down, you're checking if this is defined. That means that this should be optional, probably?

Same in `recorderApp`.